### ISSUE TITLE
Read All functionality

### ISSUE THREAD

**Issue by s7anley** [mnapoli/externals#31]
Hi,

firstly, thanks for externals! I started using it today, but I already see need for some functionality which hide or mark threads as read. There are multiple ways e.g. read all, read all loaded, read all selected or hide (ignore) threads. What you think?

**Comment by mnapoli** [comment]
Hi, yes it would be possible, maybe the best course of action would be a button on the home page to mark a thread as read. Marking all threads as read will create *a lot* of data in the database I think (the way it's all stored is not really optimized to be honest).

**Comment by staabm** [comment]
I guess it would be worthwile to mark all as read of the current page (so all which are visible to the user atm as unread.

It doesnt make much sense to mark mails/threads as read which are months/years old

**Comment by s7anley** [comment]
I agree, read all history doesn't make sense.  Regarding the solution, I mentioned multiple options in initial issue description, so now it's more about picking only one. For my use-case, I would second the `read of the current page` solution.

**Comment by mnapoli** [comment]
That sounds reasonable indeed!

**Comment by s7anley** [comment]
Ok, I will start working on it then.
